core/services/workflows/engine.go

```diff
@@ -24,23 +21,14 @@ const (
 
 type Engine struct {
 	services.StateMachine
-	logger          logger.Logger
-	registry        types.CapabilitiesRegistry
-	triggerType     string
-	triggerConfig   *values.Map
-	trigger         capabilities.TriggerCapability
-	consensusType   string
-	consensusConfig *values.Map
-	consensus       capabilities.ConsensusCapability
-	targets         []target
-	callbackCh      chan capabilities.CapabilityResponse
-	cancel          func()
-}
-
-type target struct {
-	typeStr    string
-	config     *values.Map
-	capability capabilities.TargetCapability
+	logger     logger.Logger
+	registry   types.CapabilitiesRegistry
+	trigger    capabilities.TriggerCapability
```

Comment:
nit: perhaps out of scope, but there is a lot of stuttering. capabilities.XXXCapability / capabilities.CapibilityZZZ should be renamed to capabiilties.XXX/ZZZ

Reply:
I agree, but going to leave this out of scope for this PR. I'll follow up though